server/api/interest/interest.controller.js

```diff
@@ -102,25 +104,51 @@ const createInterest = async (req, res) => {
 
 const updateInterest = async (req, res) => {
   try {
-    const interest = req.body
-    const updates = {
-      // this ... conditionally adds the $set to the update if value is present
-      ...(interest.status && { $set: { status: interest.status } }),
-      ...(interest.messages && { $push: { messages: interest.messages } })
-    }
-    const result = await Interest
-      .accessibleBy(req.ability, Action.UPDATE)
-      .updateOne(req.params, updates)
-    if (result.nModified === 0) {
+    const existingInterest = await Interest.findOne(req.params)
+      .accessibleBy(req.ability, Action.READ)
+
+    if (existingInterest === null) {
       return res.sendStatus(404)
     }
 
-    const interestDetail = await getInterestDetail(req.params._id)
-    interestDetail.type = interest.type
-    if (!interest.type) {
-      console.log('no iterest type', interest)
+    const interestUpdateData = req.body
+    const person = req.session.me
+
+    if (
+      interestUpdateData.status &&
+      person.role.includes(Role.VOLUNTEER_PROVIDER) &&
+      existingInterest.person.toString() === req.session.me._id.toString()
+    ) {
+      if (!isValidTransition(existingInterest.status, interestUpdateData.status)) {
+        return res.status(403).json({
+          message: 'Invalid status transition'
+        })
+      }
+    }
+
+    if (interestUpdateData.status) {
+      existingInterest.status = interestUpdateData.status
+    }
+
+    if (interestUpdateData.messages) {
+      interestUpdateData.messages = Array.isArray(interestUpdateData.messages)
+        ? interestUpdateData.messages : [interestUpdateData.messages]
```

Comment:
messages should always be an array. even if usually it contains only one message. but this is a safe action.

Reply:
> messages should always be an array. even if usually it contains only one message. but this is a safe action.

I found code & comments that contradict that -

Some of the tests in `server/api/interest/__tests__/interest.ability.spec.js` send messages as an array, e.g.

https://github.com/voluntarily/vly2/blob/9a702d5e80b02ae96f1dc140a093f23718b61b63/server/api/interest/__tests__/interest.ability.spec.js#L206-L209

And one of the tests in `server/api/interest/__tests__/interest.spec.js` has a comment mentioning it could be either:

https://github.com/voluntarily/vly2/blob/9a702d5e80b02ae96f1dc140a093f23718b61b63/server/api/interest/__tests__/interest.spec.js#L216-L219

If the intent is for it to always be an array maybe we should update these areas to match that?